[Tng Wen Xi] iP

wx-03 opened the pull request.

wx-03 committed:
Rename and implement greeting

wx-03 committed:
Implement echo

wx-03 committed:
Implement add and list

wx-03 committed:
Implement mark and unmark

wx-03 committed:
Use a class to represent tasks

wx-03 committed:
Edit code to comply with java coding standards

wx-03 committed:
Refactor code by shifting code from main into sendMessage() method.

wx-03 committed:
Refactor code to improve code quality and readability

wx-03 committed:
Add ToDos, Events, and Deadlines

wx-03 committed:
Add automated text UI testing

Bev-low commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,193 @@
+        }
+    }
+
+    private static void sendMessage(String userInput) {
+        String[] inputArr = userInput.split(" ");
+        String command = inputArr[0];
+
+        if (command.equalsIgnoreCase("bye")) {
+            executeCommand(CommandType.BYE, inputArr);
+        } else if (command.equalsIgnoreCase("list")) {
+            executeCommand(CommandType.LIST, inputArr);
+        } else if (command.equalsIgnoreCase("mark")) {
+            executeCommand(CommandType.MARK, inputArr);
+        } else if (command.equalsIgnoreCase("unmark")) {
+            executeCommand(CommandType.UNMARK, inputArr);
+        } else if (command.equalsIgnoreCase("todo")){
+            executeCommand(CommandType.TODO, inputArr);
+        } else if (command.equalsIgnoreCase("deadline")){
+            executeCommand(CommandType.DEADLINE, inputArr);
+        } else if (command.equalsIgnoreCase("event")){
+            executeCommand(CommandType.EVENT, inputArr);
+        }
+    }
```
Is there a better way to do this other than nested else if?

Bev-low commented on src/main/java/CommandType.java:
```diff
@@ -0,0 +1,9 @@
+public enum CommandType {
```
Good use of Enums here, makes the code quite clean

Bev-low commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,193 @@
+            break;
+
+        case EVENT:
+            for (int i = 0; i < args.length; i++) {
+                if (args[i].startsWith("/from")) {
+                    fromIdx = i;
+                } else if (args[i].startsWith("/to")) {
+                    toIdx = i;
+                }
+            }
+            if (fromIdx == -1) {
+                printReply("From when???");
+                break;
+            }
+            if (toIdx == -1) {
+                printReply("To when???");
+                break;
+            }
```
These if statments can be taken out and replaces with Enums

Bev-low commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,193 @@
+                printReply("Please input a number");
+                break;
+            }
+            if (idx < 0 || idx >= size) {
+                printReply("Invalid index");
+                break;
+            }
+            tasks[idx].setAsDone();
+            type = tasks[idx].getClass();
+            t = 'U';
+            if (type.equals(Deadline.class)) {
+                t = 'D';
+            } else if (type.equals(ToDo.class)) {
+                t = 'T';
+            } else if (type.equals(Event.class)) {
+                t = 'E';
+            }
```
These if statements also can be removed and replace with Enums

wx-03 committed:
Refactor code by overloading toString() method and extracting method to print task count.

Currently, task type needs to be checked before printing. Overload the
toString() method in each class so that the task can be printed
directly.

wx-03 committed:
Add error handling using exceptions

wx-03 committed:
Merge branch 'branch-Level-5'

wx-03 committed:
Divide classes into packages

wx-03 committed:
Merge branch 'branch-A-packages'

wx-03 committed:
Update text UI test files

wx-03 committed:
Use ArrayList to store tasks instead of fixed-size array

wx-03 committed:
Add support for deleting tasks from the list

wx-03 committed:
Save tasks on exit and load saved tasks on startup

wx-03 committed:
Merge branch 'branch-Level-6'

wx-03 committed:
Merge branch 'branch-Level-7'

wx-03 committed:
Create JAR file and fix bugs

tzerbin commented on src/main/java/luke/Luke.java:
```diff
@@ -0,0 +1,301 @@
+
+public class Luke {
+
+    private static final String LOGO =
```
Very good work in using constants to prevent the use of magic strings and numbers!

tzerbin commented on src/main/java/luke/Luke.java:
```diff
@@ -0,0 +1,301 @@
+        fw.close();
+    }
+
+    private static void executeCommand(CommandType commandType, String[] inputArr) {
```
This is a very long method. Perhaps extract functionalities out for each command so as to allow your code to be more readable?

tzerbin replied:
Would you want to name your parameter as `inputArr` instead of `inputs`?

tzerbin commented on src/main/java/luke/Luke.java:
```diff
@@ -0,0 +1,301 @@
+        String[] inputArr = userInput.split(" ");
+        String command = inputArr[0];
+
+        if (command.equalsIgnoreCase("bye")) {
```
This is a very long piece of code. You may want to explore inheritance together with polymorphism to cut down the amount of code here.

tzerbin commented on src/main/java/luke/Luke.java:
```diff
@@ -0,0 +1,301 @@
+    private static void loadSingleTask(String taskStr) {
+        String[] taskStrArr = taskStr.split("\\|");
+        switch (taskStrArr[TASK_TYPE_INDEX]) {
+        case "T" -> tasks.add(new ToDo(taskStrArr[DESCRIPTION_INDEX], taskStrArr[ISDONE_INDEX].equals("1")));
```
To prevent error-prone code, perhaps you should use colons ':' instead of arrow "->" instead?
You can also use default branches.

tzerbin commented on src/main/java/luke/Luke.java:
```diff
@@ -0,0 +1,301 @@
+        if (!dataDir.exists()) {
+            dataDir.mkdir();
+        }
+        File save = new File("data/tasks.txt");
```
There are still quite a number of magic strings lying around the codebase. Please take a look!

tzerbin commented on src/main/java/luke/exceptions/IncorrectInput.java:
```diff
@@ -0,0 +1,7 @@
+package luke.exceptions;
+
+public class IncorrectInput extends LukeException {
```
You may want to name your classes with a postfix "Exception" that are extending from LukeException?

tzerbin commented on src/main/java/luke/tasks/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+
+public class Deadline extends Task{
+    protected String by;
+
```
There is an excessive use of whitespace here.
(This is part of the intermediate portion of coding standard.)

tzerbin commented on src/main/java/luke/tasks/Event.java:
```diff
@@ -0,0 +1,35 @@
+    public String toString() {
+        return String.format("[E][%s] %s (from: %s to: %s)", this.isDone ? "X" : " ", this.description, from, to);
+    }
+    public String toString2() {
```
Could you rename the method with a more descriptive name, instead of labelling methods with integer postfix?

tzerbin reviewed (commented):
Good attempt in following the guidelines, especially on the attempt of removing magic strings.
Some nits here and there!

wx-03 committed:
Update text UI test to delete data directory if it exists before running test

wx-03 committed:
Update .gitignore

wx-03 committed:
Add Ui class

wx-03 committed:
Add Storage and TaskList classes, refactor code

wx-03 committed:
Fix bug

Currently, FileAlreadyExistsException is thrown when running the bot after the first time. Fix this by creating the directory and data file only if they do not exist.

wx-03 committed:
Move all task operation methods to TaskList class

wx-03 committed:
Add Parser and Command classes

wx-03 committed:
Merge branch 'branch-A-MoreOOP'

wx-03 committed:
Move Command classes to commands package

wx-03 committed:
Remove commented-out code

wx-03 committed:
Rename variable from inputArr to inputs

wx-03 committed:
Remove unused method

wx-03 committed:
Refactor code

Remove unused method and rename exception classes to include postfix "Exception"

wx-03 committed:
Remove excessive whitespace

wx-03 committed:
Rename methods labelled with integer postfix

wx-03 committed:
Add FindCommand class

wx-03 committed:
Refactor code by extracting getArgs method

wx-03 committed:
Implement find command

wx-03 committed:
Update text ui test

wx-03 committed:
Change filepath to final in Storage class

wx-03 committed:
Change task ArrayList from public to private in TaskList class

wx-03 committed:
Add JavaDoc comments

wx-03 committed:
Store deadline as a java.time.LocalDateTime instead of a string

wx-03 committed:
Save event start and end timings as java.time.LocalDateTime instead of String

wx-03 committed:
Merge pull request #2 from wx-03/branch-Level-9

Branch level 9

wx-03 committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/luke/Storage.java
#	src/main/java/luke/TaskList.java

wx-03 committed:
Merge pull request #3 from wx-03/branch-A-JavaDoc

Add JavaDoc comments

wx-03 committed:
Merge branch 'master' into branch-Level-8

# Conflicts:
#	src/main/java/luke/TaskList.java

wx-03 committed:
Merge pull request #1 from wx-03/branch-Level-8

Branch level 8

wx-03 committed:
Trim whitespace from user input

wx-03 committed:
Allow find command to search for queries with more than 1 word

wx-03 committed:
Update README.md

wx-03 committed:
Update text ui test
